src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java

```diff
@@ -49,6 +58,42 @@ public class XdocsTemplateSink extends XdocSink {
     public XdocsTemplateSink(Writer writer, String encoding) {
         super(writer);
         this.encoding = encoding;
+        this.writer = new PrintWriter(writer);
+    }
+
+    /**
+     * Customization of org.apache.maven.doxia.sink.impl.XhtmlBaseSink.unifyEOL() method.
+     *
+     * @param text the text to scan.
+     * @return a String that contains only Linux System EOLs.
+     */
+    private static String unifyToLinuxEndOfLine(String text) {
+        return text
+                .replace("\r\n", LINUX_EOL)
+                .replace('\r', LINUX_EOL.charAt(0));
```

Comment:
We don't need a method for this, we can just inline replacement in `write`, below.

Reply:
it was created to "match" what original implementation is doing
https://github.com/apache/maven-doxia/blob/240f2b7ffe00c29446f6f6006327ff9394e02043/doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java#L2006-L2014

Reply:
but this method is protected - https://github.com/apache/maven-doxia/blob/245a5f8364b179673acda6c896b0d4aa7381cb31/doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/AbstractSink.java#L41 , why we did not override it ?

Reply:
@kalpadiptyaroy , ^^^ please remind why we did not override `protected static String unifyEOLs(String text) {` ?

Reply:
@romani you can't override a static method.